packages/legend-application-studio/src/stores/editor-state/element-editor-state/mapping/MappingElementDecorator.ts

```diff
@@ -439,13 +426,20 @@ export class MappingElementDecorator implements SetImplementationVisitor<void> {
         setImplementation,
         decoratePropertyMapping,
       );
-    mapping_setPropertyMappings(
+    instanceSetImpl_setPropertyMappings(
       setImplementation,
       decoratedPropertyMappings.concat(
-        propertyMappingsBeforeDecoration.filter(
-          (propertyMapping) =>
-            !decoratedPropertyMappings.includes(propertyMapping),
-        ),
+        propertyMappingsBeforeDecoration
+          .filter(
```

Comment:
I would like us to document this a bit, this is a fair optimization, but in the same spirit as this PR, we might in the future would like to pertain these properties instead